pkg/util/kubernetes/apiserver/leaderelection/leaderelection.go

```diff
@@ -230,6 +232,45 @@ func (le *LeaderEngine) IsLeader() bool {
 	return le.GetLeader() == le.HolderIdentity
 }
 
+// reportLeaderMetric updates the value of the metric is_leader every 15 seconds
+func (le *LeaderEngine) reportLeaderMetric(ctx context.Context) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	isLeader := telemetry.NewGaugeWithOpts(
+		"leader_election",
+		"is_leader",
+		[]string{"join_leader", "is_leader"}, // join_leader is for label joins
+		"The label is_leader is true if the reporting pod is leader, equals false otherwise.",
+		telemetry.Options{NoDoubleUnderscoreSep: true},
+	)
+
+	ticker := time.NewTicker(15 * time.Second)
+	defer ticker.Stop()
+
+	log.Info("Start reporting leader metric")
+	defer log.Info("Stop reporting leader metric")
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			// We want to make sure only one (the latest) context is exposed for this metric
+			// Delete previous run metric
+			isLeader.Delete("true", "false")
+			isLeader.Delete("true", "true")
+
+			isLeaderLabel := "false"
+			if le.IsLeader() {
+				isLeaderLabel = "true"
+			}
+
+			isLeader.Set(1.0, "true", isLeaderLabel) // join_leader is always true for label joins
```

Comment:
constant value `1.0` because of https://github.com/DataDog/integrations-core/blob/master/datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py#L529-L535